presto-main/src/main/java/com/facebook/presto/security/AccessControl.java

```diff
@@ -165,18 +167,29 @@
     void checkCanCreateViewWithSelectFromView(TransactionId transactionId, Identity identity, QualifiedObjectName viewName);
 
     /**
-     * Check if identity is allowed to grant a privilege on the specified table.
+     * Check if identity is allowed to grant the specified privilege on the specified table.
      * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
      */
     void checkCanGrantTablePrivilege(TransactionId transactionId, Identity identity, Privilege privilege, QualifiedObjectName tableName);
 
     /**
-     * Check if identity is allowed to revoke a privilege on the specified table.
+     * Check if identity is allowed to revoke the specified privilege on the specified table.
      * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
      */
     void checkCanRevokeTablePrivilege(TransactionId transactionId, Identity identity, Privilege privilege, QualifiedObjectName tableName);
 
     /**
+     * Check if identity is allowed to show grants on the specified catalog or schema or table.
+     * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
+     */
+    void checkCanShowGrants(TransactionId transactionId, Identity identity, QualifiedTablePrefix prefix);
```

Comment:
Also, when we added `SHOW TABLES` we wanted to only show tables you had grants on, which is how the spec says the information schema table it is based on should work, but didn't know how to do that at the time.  My point is these APIs are really linked (and it would be nice to align to the spec in the future).